From aae023064e8ace77354a6f6136cdb3bcbefbc2eb Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 31 Mar 2023 20:13:33 +0200
Subject: [PATCH 0103/1141] dt-bindings: dma: stm32-dma3: introduce
 st,axi-max-burst-len property

DMA3 maximum burst length (in unit of beat) may be restricted depending
on bus interconnect.
Introduce st,axi-max-burst-len. If used, it will clamp the burst length
to that value, if not, the maximum burst length value supported by DMA3
is used.

Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: Ic4975cf40c72a86f70e9561f86dc764510fbe9e8
---
 Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
+++ b/Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
@@ -103,6 +103,14 @@ properties:
     minItems: 4
     maxItems: 16
 
+  st,axi-max-burst-len:
+    description: |
+      Restrict master AXI burst length in unit of beat by value specified in this property.
+      If this property is missing the maximum AXI burst length supported by DMA3 is used.
+    $ref: /schemas/types.yaml#/definitions/uint32
+    minimum: 1
+    maximum: 256
+
 required:
   - compatible
   - reg
@@ -136,5 +144,6 @@ examples:
                    <GIC_SPI 48 IRQ_TYPE_LEVEL_HIGH>;
       clocks = <&rcc CK_BUS_HPDMA1>;
       #dma-cells = <4>;
+      st,axi-max-burst-len = <16>;
     };
 ...
